tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,425 @@
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    });
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to oneofs", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`
+    edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    message Message {
+        oneof SomeOneOf {
+            option features.json_format = ALLOW;
+            int32 a = 13;
+            string b = 14;
+        }
+    }`).root.resolveAll();
+
+    // console.log(rootEditionsOverriden.lookup("SomeOneOf")._features)
+    test.same(rootEditionsOverriden.lookup("SomeOneOf")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'ALLOW',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to extensions", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`
+    edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    message Message {
+        extensions 10 to 100;
+        extend Message {
+            int32 bar = 10 [features.utf8_validation = NONE];
+        }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup("Message").lookup(".Message.bar")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'NONE',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to top-level enum", function(test) {
+
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+    enum SomeEnum {
+        option features.utf8_validation = NONE;
+        ONE = 1;
+        TWO = 2;
+    }`).root.resolveAll();
+
+    console.log(rootEditionsOverriden.lookup("SomeEnum")._features)
+    test.same(rootEditionsOverriden.lookup("SomeEnum")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'NONE',
+        '(abc)': { d_e: 'deeply_nested_false' } 
+    })
+
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to service and method", function(test) {
```

Comment:
nit: "service to method", the test itself is fine (adding a few file features for extra coverage is reasonable, but not the primary behavior under test)